include/PriorityQueue.h

```diff
@@ -0,0 +1,45 @@
+#pragma once
+
+#include <iostream>
+#include <list>
+#include <exception>
+
+
+template<class T>
+struct MyComparator {
+	T operator()(T a, T b) const {
+		return a - b;
+	}
+};
+
+template <class T>
+class PriorityQueue {
+   
+public:
+	
+	void push(const T& t); 
+	T poll();
+
+private:
+	std::list<T> m_queue;
+};
+
+template<class T>
+inline void PriorityQueue<T>::push(const T& t)
+{
+	auto iter = m_queue.begin();

```

Comment:
When possible it's recommended to call cbegin() and cend(). You can never be too safe